x-pack/plugins/alerting/server/routes/suggestions/values_suggestion_rules.ts

```diff
@@ -0,0 +1,109 @@
+import { getRequestAbortedSignal } from '@kbn/data-plugin/server';
+import { termsAggSuggestions } from '@kbn/unified-search-plugin/server/autocomplete/terms_agg';
+import type { ConfigSchema } from '@kbn/unified-search-plugin/config';
+import { UsageCounter } from '@kbn/usage-collection-plugin/server';
+import { getKbnServerError, reportServerError } from '@kbn/kibana-utils-plugin/server';
+import { ALERTING_CASES_SAVED_OBJECT_INDEX } from '@kbn/core-saved-objects-server/src/saved_objects_index_pattern';
+import type * as estypes from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+
+import { verifyAccessAndContext } from '../lib';
+import { ILicenseState } from '../../lib';
+import { AlertingRequestHandlerContext } from '../../types';
+import {
+  AlertingAuthorizationEntity,
+  AlertingAuthorizationFilterOpts,
+  AlertingAuthorizationFilterType,
+} from '../../authorization';
+import { RuleAuditAction, ruleAuditEvent } from '../../rules_client/common/audit_events';
+
+const alertingAuthorizationFilterOpts: AlertingAuthorizationFilterOpts = {
+  type: AlertingAuthorizationFilterType.ESDSL,
+  fieldNames: { ruleTypeId: 'alert.alertTypeId', consumer: 'alert.consumer' },
+};
+
+export const RulesSuggestionsSchema = {
+  body: schema.object(
+    {
+      field: schema.string(),
+      query: schema.string(),
+      filters: schema.maybe(schema.any()),
+      fieldMeta: schema.maybe(schema.any()),
+    },
+    { unknowns: 'allow' }
+  ),
+};
+
+export function registerValueSuggestionsRoute(
+  router: IRouter<AlertingRequestHandlerContext>,
+  licenseState: ILicenseState,
+  config$: Observable<ConfigSchema>,
+  usageCounter?: UsageCounter
+) {
+  router.post(
+    {
+      path: '/internal/rules/suggestions/values',
+      validate: RulesSuggestionsSchema,
+    },
+    router.handleLegacyErrors(
+      verifyAccessAndContext(licenseState, async function (context, request, response) {
+        const config = await firstValueFrom(config$);
+        const { field: fieldName, query, fieldMeta } = request.body;
+        const abortSignal = getRequestAbortedSignal(request.events.aborted$);
+        const { savedObjects, elasticsearch } = await context.core;
+
+        const rulesClient = (await context.alerting).getRulesClient();
+        let authorizationTuple;
+        try {
+          authorizationTuple = await rulesClient
+            .getAuthorization()
+            .getFindAuthorizationFilter(
+              AlertingAuthorizationEntity.Rule,
+              alertingAuthorizationFilterOpts
+            );
+        } catch (error) {
+          rulesClient.getAuditLogger()?.log(
+            ruleAuditEvent({
+              action: RuleAuditAction.FIND,
+              error,
+            })
+          );
+          throw error;
+        }
+
+        const { filter: authorizationFilter } = authorizationTuple;
+        const filters = [
+          ...(authorizationFilter != null ? [authorizationFilter] : []),
```

Comment:
Are we using `!=` because `authorizationFilter` might also be `undefined` or is its only falsy value `null`? I'd prefer we `!==` if there isn't an explicit reason not to because at the end of the day we are still indeed writing in a toy programming language cobbled together in 10 sleepless days in 1995 by a man ousted from Mozilla for homophobia and it is PRIDE MONTH so we are SAYING NO TO BRENDAN EICH IN OUR CODING STYLE